### PR TITLE
Implement basic enc, decoding

### PR DESCRIPTION
**Pull request by A6GibKm**
I am not very happy with having the decrypt in item.rs and the decrypt in secret.rs

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,10 +11,19 @@ serde = "1.0"
...
+zeroize = "1.5"
 
 [[bin]]
 name = "basic"
 path = "examples/basic.rs"
 
 [dev-dependencies]
 serde_json = "1.0"
+
+[dependencies.cipher]
```
Comment (bilelmoussaoui) [inline Cargo.toml:27]: I would inline it like the other dependencies above

File: src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+use aes::cipher::{
+    block_padding::Pkcs7, BlockDecryptMut, BlockEncryptMut, BlockSizeUser, KeyIvInit,
+};
+use hmac::{digest::OutputSizeUser, Mac};
+use zeroize::Zeroizing;
+
+use crate::error::Result;
```
Comment (bilelmoussaoui) [inline src/utils.rs:7]:
```suggestion
use crate::Result;
```

File: src/dbus/api/secret.rs [hunk 3]
```diff
@@ -26,12 +26,9 @@ impl<'a> Secret<'a> {
...
     ) -> Self {
-        let (parameters, value) = match algorithm.as_ref() {
-            Algorithm::Plain => (vec![], secret.to_vec()),
-            Algorithm::Dh(_blob) => {
-                // See https://github.com/hwchen/secret-service-rs/blob/d6aaa774f0ec504ff5f26662279e07175b8ef111/src/util.rs#L52
-                unimplemented!()
-            }
+        let (value, parameters) = match algorithm.as_ref() {
+            Algorithm::Plain => (secret.to_vec(), vec![]),
+            Algorithm::Dh(aes_key) => utils::encrypt(secret, aes_key).unwrap(),
```
Comment (bilelmoussaoui) [inline src/dbus/api/secret.rs:31]: The key should no longer be stored inside Dh variant, I would also maybe just rename it to Encrypted  and we could document what encryption it uses in the future. The key then should be generated and stored as part of src/dbus/service.rs and passed to the src::api::Session when created.

File: Cargo.toml (hunk 1 above)
Comment (A6GibKm) [reply Cargo.toml]: Done.

File: src/dbus/api/secret.rs (hunk 3 above)
Comment (A6GibKm) [reply src/dbus/api/secret.rs]: Renamed to Encrypted, without a inner field. With this one can drop all the `Arc`s required. The key should be retrievable from the session right?

Comment (bilelmoussaoui) [reply src/dbus/api/secret.rs]: I moved it to src/dbus, so it probably needs a rebase here

**Pull request closed without merging**